File: WeatherLoader/src/main/java/org/arqsoft/WeatherLoader/domain/model/Weather.java
package org.arqsoft.WeatherLoader.domain.model;
import lombok.Data;
import org.springframework.data.mongodb.core.mapping.Document;
import java.time.LocalDateTime;

@Data
@Document
public class Weather {
    private LocalDateTime date;
    private Float temperature;
    private Float sensation;
    private Float humidity;
    private Location location;
}

File: WeatherMetric/src/main/kotlin/ar/edu/unq/weather/metric/domain/Weather.kt
package ar.edu.unq.weather.metric.domain

import java.time.LocalDateTime

data class Weather(
        val temperature: Float,
        val unit: Unit,
        val sensation: Float,
        val humidity: Float,
        val locality: Locality,
        val date: LocalDateTime
)

enum class Unit {
    FAHRENHEIT, CELSIUS, KELVIN;
}


interface ILoaderService {
    fun currentWeather(locality: Locality, unit: Unit): Weather
    fun avgLastWeek(locality: Locality, unit: Unit): Weather
    fun lastWeekTemps(locality: Locality, unit: Unit): List<Weather>
    fun avgLastDay(locality: Locality, unit: Unit): Weather

}